### PR TITLE
Unify error type

### PR DESCRIPTION
**Pull request by SirVer**
Consolidate client and server errors into one type. There are still separate errors for LuaDictionaries, individual plugins and RPCs - because they just seem too distinct to merge into the main error.

### CODE REVIEW

**Comment by SirVer** [comment]
@SShrike Wanna take this code review? 

@gchp ping fyi. This is the branch we started on together.

**Comment by gchp** [comment]
Looks good to me, nice job!

**Comment by SShrike** [comment]
I'm probably not the best person for code reviews since I'm a complete Rust novice (haven't even finished the Rust book haha)

Once I acquaint myself with Rust I'd be happy to do some code reviews, although be warned I'm new to all this. (only been into programming for approximately a year and a half)

**Comment by SirVer** [comment]
@SShrike I always did open source because I find it gets stuff done fastest AND because it is a great vehicle to learn. I learned all I know about programming working on my first open source project Widelands and many contributors on Widelands started coding on it. Code reviews are super useful for that - and beneficial for both the writer and the reader. 

First, code always gets better through a second pair of eyes - if somebody does not understand something in a code review, it is likely that it is too complicated. If the coder is more experienced than the reviewer, the reviewer will see live code 'growing' and have a good idea of the thought process - which is more meaningful than just finished code. He/She will also see nifty tricks.

If it is the other way around, the coder will get direct feedback that will let them grow and improve the code base. More often than not though, the coder and reviewer have disjunct knowledge and both will learn something from the review, i.e. both will be more experienced and less experienced at the same time.

tl/dr: just take a code review and read through the code next time. Ask questions and try to get the most out of it. Volunteer for code reviews in other open source projects - reviewers are always sought after. 

For now I am going to merge this as Greg already provided his second pair of eyes for this. 

@homu  r+
